src/states/client.rs

```diff
@@ -200,7 +210,40 @@ impl Client {
             return Ok(Transition::Stay);
         }
 
-        Ok(self.dispatch_routing_message(routing_msg.clone(), outbox))
+        Ok(self.dispatch_routing_message(routing_msg, outbox))
+    }
+
+    fn handle_direct_message(&mut self,
+                             direct_msg: DirectMessage,
+                             outbox: &mut EventBox)
+                             -> Result<Transition, RoutingError> {
+        use messages::DirectMessage::*;
+        match direct_msg {
```

Comment:
yes, I do agree there are couple of places of in-consistent approach of the `explicit list of full variants`
probably we shall have a separate PR to make the approach consistent.
anyway, this this just a trival comment and shall not block the PR to be merged.